service/matching/ratelimit_manager.go

```diff
@@ -113,10 +168,15 @@ func (r *rateLimitManager) InjectWorkerRPS(meta *pollMetadata) {
 			rps = &value
 		}
 	}
+	// Short-circuit if there's no change from the previous worker RPS.
```

Comment:
no, let's not add complicated logic to do extra short-circuits. there should be exactly one short-circuit check: when we have the new effective number, compare it to the last, and only update the actual limiters if it changed.